tests/lib/azure/fixture/fixture.go

```diff
@@ -0,0 +1,60 @@
+package fixture
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/NeowayLabs/klb/tests/lib/azure"
+)
+
+// Fixture provides you the basic data to write your tests, enjoy :-)
+type F struct {
+	//ResGroupName is the resource group name where
+	//all resources will be created
+	ResGroupName string
+	//Session used to interact with the API
+	Session *azure.Session
+	//Location where resources are created
+	Location string
+}
+
+type Test func(*testing.T, F)
+
+// Run creates a unique resource group based on testname and calls
+// the given testfunc passing as argument all the resources required
+// to test integration with Azure cloud.
+//
+// It's main purpose is to make your life easier and guarantee that
+// the resource group will be destroyed when testfunc exits.
+// It is a programming error to reference the created resource group
+// after returning from testfunc (just like Go http handlers).
+//
+// Since testing is pretty slow and the fixture guarantee unique named
+// resource groups it will also run the your test in parallel with others
+// so you don't have to die waiting for a result.
+func Run(
+	t *testing.T,
+	testname string,
+	location string,
+	testfunc Test,
+) {
+	t.Run(testname, func(t *testing.T) {
+		t.Parallel()
+
+		session := azure.NewSession(t)
+		resgroup := fmt.Sprintf("klb-test-%s-%d", testname, rand.Intn(9999999))
+
+		resources := azure.NewResourceGroup(t, session)
+		defer resources.Delete(t, resgroup)
+
+		resources.Create(t, resgroup, location)
+		resources.AssertExists(t, resgroup)
```

Comment:
The FatalNow code:

```
// FailNow marks the function as having failed and stops its execution.
// Execution will continue at the next test or benchmark.
// FailNow must be called from the goroutine running the
// test or benchmark function, not from other goroutines
// created during the test. Calling FailNow does not stop
// those other goroutines.
func (c *common) FailNow() {
	c.Fail()

	// Calling runtime.Goexit will exit the goroutine, which
	// will run the deferred functions in this goroutine,
	// which will eventually run the deferred lines in tRunner,
	// which will signal to the test loop that this test is done.
	//
	// A previous version of this code said:
	//
	//	c.duration = ...
	//	c.signal <- c.self
	//	runtime.Goexit()
	//
	// This previous version duplicated code (those lines are in
	// tRunner no matter what), but worse the goroutine teardown
	// implicit in runtime.Goexit was not guaranteed to complete
	// before the test exited. If a test deferred an important cleanup
	// function (like removing temporary files), there was no guarantee
	// it would run on a test failure. Because we send on c.signal during
	// a top-of-stack deferred function now, we know that the send
	// only happens after any other stacked defers have completed.
	c.finished = true
	runtime.Goexit()
}
```

It seems good enough for us, we just need to pay attention to never call t.Fatal from some goroutine we create on the tests.